File: !OLD/converter_data_cloud.py
import json
import re
from pathlib import Path

ARQUIVO_ORIGINAL = Path("data.js")
ARQUIVO_NOVO = Path("data_cloud.js")

CLOUD_NAME = "dz3uocw2h"
PASTA_CLOUDINARY = "arquivo-desfiles"

def carregar_data_js(caminho):
    texto = caminho.read_text(encoding="utf-8")
    texto = re.sub(r"^\s*const\s+data\s*=\s*", "", texto)
    texto = re.sub(r";\s*$", "", texto)
    return json.loads(texto)

def salvar_data_js(caminho, data):
    conteudo = "const data = " + json.dumps(data, indent=2, ensure_ascii=False) + ";\n"
    caminho.write_text(conteudo, encoding="utf-8")

data = carregar_data_js(ARQUIVO_ORIGINAL)

for item in data:
    imagem = item.get("image", "")

    if imagem.startswith("images/"):
        caminho_sem_images = imagem.replace("images/", "", 1)
        item["image"] = (
            f"https://res.cloudinary.com/{CLOUD_NAME}/image/upload/"
            f"{PASTA_CLOUDINARY}/{caminho_sem_images}"
        )

salvar_data_js(ARQUIVO_NOVO, data)

print(f"Pronto: {ARQUIVO_NOVO} criado.")
print(f"Total de itens: {len(data)}")

File: !OLD/converter_data_cloud.js
const fs = require("fs");

const CLOUD_NAME = "dz3uocw2h";
const PASTA_CLOUDINARY = "arquivo-desfiles";

const texto = fs.readFileSync("data.js", "utf8");

const codigo = texto + "\nmodule.exports = data;";
const modulo = { exports: null };

const func = new Function("module", codigo);
func(modulo);

const data = modulo.exports;

data.forEach(item => {
  if (item.image && item.image.startsWith("images/")) {
    const caminho = item.image.replace("images/", "");
    item.image = `https://res.cloudinary.com/${CLOUD_NAME}/image/upload/${PASTA_CLOUDINARY}/${caminho}`;
  }
});

const novoArquivo = "const data = " + JSON.stringify(data, null, 2) + ";\n";

fs.writeFileSync("data_cloud.js", novoArquivo, "utf8");

console.log("Pronto: data_cloud.js criado");
console.log("Total de itens:", data.length);